main.mo

```diff
@@ -95,6 +96,17 @@ shared({caller}) actor class Rex(
         Events.Events.remove(events_, token, index);
     };
 
+    public shared({caller}) func balances() : async [(Principal, Ledger.Tokens)] {
```

Comment:
Can be a query method, no?

Reply:
No, query methods do not have 'sending'-capabilities.
We need to make it an update method to be able to query the ledger.